.github/workflows/release-cycle.yml

```diff
@@ -0,0 +1,35 @@
+name: Test release cycle
+
+on: [pull_request, push, workflow_dispatch]
+
+env:
+  FORCE_COLOR: 1
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest]
+
+    steps:
+      - name: Silence Windows warnings
+        run: |
+          git config --global core.eol lf
+          git config --global core.autocrlf input
+
```

Comment:
```suggestion
```

I didn't finish my review before you noticed the same issue, that the config was not getting set _before_ checkout, but instead of setting custom config, we can just do what local users will and actually check the staged index against the repo `HEAD` instead of the working tree against the index/repo `HEAD`. That will verify, per @merwok 's question, that users won't actually face a problem, and is also simpler and less finicky.

Reply:
Much cleaner, thank you!

https://github.com/python/devguide/actions/runs/3580947474/jobs/6023523022